### PR TITLE
Read new key license_v2 from SUBNET response

### PR DESCRIPTION
**Pull request by anjalshireesh**
## Community Contribution License

All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license] (https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description

SUBNET now has a v2 of license that is returned in the new key `license_v2`. mc will start reading and storing the same.
(The old key `license` is deprecated but is still available in SUBNET response to ensure that the current released version of mc doesn't break)

## Motivation and Context

Stay up-to-date with license changes in SUBNET

## How to test this PR?

Following commands should continue to work fine:

- mc license register
- mc license update
- mc support diag
- mc support perf
- mc support profile
- mc  support inspect

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -124,9 +126,11 @@ require (
...
 	golang.org/x/sync v0.3.0 // indirect
-	golang.org/x/sys v0.13.0
+	golang.org/x/sys v0.15.0
 	google.golang.org/genproto v0.0.0-20230913181813-007df8e322eb // indirect
 	google.golang.org/grpc v1.58.3 // indirect
 	google.golang.org/protobuf v1.31.0 // indirect
 	gopkg.in/ini.v1 v1.67.0 // indirect
 )
+
+replace github.com/minio/pkg/v2 => github.com/anjalshireesh/pkg/v2 v2.0.0-20231213135926-9362a8740dc3
```
Comment (anjalshireesh) [inline go.mod:136]: To be removed after https://github.com/minio/pkg/pull/90 is merged

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,8 @@
 module github.com/minio/mc
 
-go 1.19
+go 1.21
+
+toolchain go1.21.5
```
Comment (klauspost) [inline go.mod:5]: Revert.

This is not a "desired version", this is the minimum version, and it should not be 1.21

Comment (anjalshireesh) [reply go.mod]: thanks, removed.

**Review by shtripat — APPROVED** [review]
LGTM. Verified.

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**